Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -416,39 +494,22 @@ void __stdcall ShipDestroyed(DamageList *_dmg, DWORD *ecx, uint kill)
 				{
 					uint iDockedClientID = HkGetClientIdFromCharname(i->second);
 
-					// Make sure that each docked ship agrees that it's docked before sending a teleport order
-					if(!mobiledockClients[iDockedClientID].mobileDocked)
-					{
-						mobiledockClients[client].mapDockedShips.erase(i->first);
-						continue;
-					}
-
 					// Update the coordinates the given ship should launch to.
 					UpdateCarrierLocationInformation(iDockedClientID, cship->get_position(), cship->get_orientation());
 
+					// Carrier is no more. Remove character name.

```

Comment:
this feels like a weird hack

Reply:
Now plugin uses flag to check if ship docked on dead carrier.